etc/release.sh

```diff
@@ -0,0 +1,208 @@
+#
+# Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at
+# http://oss.oracle.com/licenses/upl.
+#
+#!/bin/bash
+
+# -----------------------------------------------------------------------------
+# Below mentioned environment variables must be set before invoking this script.
+# 1) BRANCH_NAME           : Name of the GIT branch to be used to run release.
+# 2) RELEASE_VERSION       : Release version to use.
+# 3) NEXT_SNAPSHOT_VERSION : Next snapshot version to use.
+# 4) RELEASE_IMAGE_PREFIX  : Docker repository prefix to be used for
+#                            Coherence Operator docker image.
+# 5) DRY_RUN               : To indicate whether to run script in dry mode.
+# -----------------------------------------------------------------------------
+
+# -----------------------------------------------------------------------------
+# Setup Release Branch
+# -----------------------------------------------------------------------------
+setupReleaseBranch()
+  {
+  echo "RELEASE_IMAGE_PREFIX = ${RELEASE_IMAGE_PREFIX}"
+
+  if [ -e ../settings.xml ]; then
+    local MVN_SETTINGS="-s ../settings.xml"
+  else
+    local MVN_SETTINGS=""
+  fi
+
+  if [ -z $(git ls-remote -q --tags | grep $RELEASE_TAG) ]; then
+    mvn $MVN_SETTINGS -DnewVersion=$RELEASE_VERSION versions:set versions:commit
+
+    pwd
+    ls -ls
+    sed "s|\${test.image.prefix}|${RELEASE_IMAGE_PREFIX}|" pom.xml
+
+    git commit -a -m "Preparing for Release Version $RELEASE_VERSION"
+    git tag $RELEASE_TAG
+
+    if [ "false" = "$DRY_RUN" ]; then
+      git push origin $RELEASE_TAG
+    fi
+
+    if [ 0 -eq $? ]; then
+      mvn -s $MVN_SETTINGS -DnewVersion=$NEXT_SNAPSHOT_VERSION versions:set versions:commit
+      if [ -n "${RELEASE_IMAGE_PREFIX}" ]; then
+        sed "s|${RELEASE_IMAGE_PREFIX}|\${test.image.prefix}|" pom.xml
+      else
+        sed "s|<release.image.prefix></release.image.prefix>|<release.image.prefix>\${test.image.prefix}</release.image.prefix>|" pom.xml
+      fi
+ 
+      git commit -a -m "Preparing for Next Development version."
+      if [ "false" = "$DRY_RUN" ]; then
+        git push
+      fi
+      return $?
+    else
+      return 1
+    fi
+  else
+    echo ""
+    echo "Git tag $RLEASE_TAG already exists so use existing release tag $RELEASE_TAG ..."
```

Comment:
$RLEASE_TAG --> $RELEASE_TAG

Reply:
Is this an error message?  I don't understand what it means...  Should it say something like: "ERROR: Git tag $RELEASE_TAG already exists" ?

Reply:
No. it's not an error message. As it says RELEASE_TAG already exist so it will use the tag instead of the pushing the tag. This can happen if release job might have failed in middle for some reason e.g. pushing docker image etc..

Reply:
Fixed.